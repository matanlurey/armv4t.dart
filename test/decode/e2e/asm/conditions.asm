FLAG_N = 1 shl 31
FLAG_Z = 1 shl 30
FLAG_C = 1 shl 29
FLAG_V = 1 shl 28

conditions:
        ; Tests for conditions

t001:
        ; EQ - Z set
        msr     cpsr_f, FLAG_Z
        beq     t002
        ; m_exit  1

t002:
        ; NE - Z clear
        msr     cpsr_f, 0
        bne     t003
        ; m_exit  2

t003:
        ; CS - C set
        msr     cpsr_f, FLAG_C
        bcs     t004
        ; m_exit  3

t004:
        ; CC - C clear
        msr     cpsr_f, 0
        bcc     t005
        ; m_exit  4

t005:
        ; MI - N set
        msr     cpsr_f, FLAG_N
        bmi     t006
        ; m_exit  5

t006:
        ; PL - N clear
        msr     cpsr_f, 0
        bpl     t007
        ; m_exit  6

t007:
        ; VS - V set
        msr     cpsr_f, FLAG_V
        bvs     t008
        ; m_exit  7

t008:
        ; VC - V clear
        msr     cpsr_f, 0
        bvc     t009
        ; m_exit  8

t009:
        ; HI - C set and Z clear
        msr     cpsr_f, FLAG_C
        bhi     t010
        ; m_exit  9

t010:
        ; LS - C clear and Z set
        msr     cpsr_f, FLAG_Z
        bls     t011
        ; m_exit  10

t011:
        ; GE - N equals V
        msr     cpsr_f, 0
        bge     t012
        ; m_exit  11

t012:
        msr     cpsr_f, FLAG_N or FLAG_V
        bge     t013
        ; m_exit  12

t013:
        ; LT - N not equals to V
        msr     cpsr_f, FLAG_N
        blt     t014
        ; m_exit  13

t014:
        msr     cpsr_f, FLAG_V
        blt     t015
        ; m_exit  14

t015:
        ; GT - Z clear and (N equals V)
        msr     cpsr_f, 0
        bgt     t016
        ; m_exit  15

t016:
        msr     cpsr_f, FLAG_N or FLAG_V
        bgt     t017
        ; m_exit  16

t017:
        ; LE - Z set or (N not equal to V)
        msr     cpsr_f, FLAG_Z
        ble     t018
        ; m_exit  17

t018:
        msr     cpsr_f, FLAG_N
        ble     t019
        ; m_exit  18

t019:
        msr     cpsr_f, FLAG_V
        ble     t020
        ; m_exit  19

t020:
        ; AL - always
        bal     conditions_passed
        ; m_exit  20

conditions_passed:
