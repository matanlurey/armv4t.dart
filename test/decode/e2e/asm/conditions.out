; Automatically generated

msr cpsr_c, 257                 ; 0xe328f101
beq 16777215                    ; 0xaffffff
msr cpsr_c, 0                   ; 0xe328f000
bne 16777215                    ; 0x1affffff
msr cpsr_c, 514                 ; 0xe328f202
bcs 16777215                    ; 0x2affffff
msr cpsr_c, 0                   ; 0xe328f000
bcc 16777215                    ; 0x3affffff
msr cpsr_c, 258                 ; 0xe328f102
bmi 16777215                    ; 0x4affffff
msr cpsr_c, 0                   ; 0xe328f000
bpl 16777215                    ; 0x5affffff
msr cpsr_c, 513                 ; 0xe328f201
bvs 16777215                    ; 0x6affffff
msr cpsr_c, 0                   ; 0xe328f000
bvc 16777215                    ; 0x7affffff
msr cpsr_c, 514                 ; 0xe328f202
bhi 16777215                    ; 0x8affffff
msr cpsr_c, 257                 ; 0xe328f101
bls 16777215                    ; 0x9affffff
msr cpsr_c, 0                   ; 0xe328f000
bge 16777215                    ; 0xaaffffff
msr cpsr_c, 521                 ; 0xe328f209
bge 16777215                    ; 0xaaffffff
msr cpsr_c, 258                 ; 0xe328f102
blt 16777215                    ; 0xbaffffff
msr cpsr_c, 513                 ; 0xe328f201
blt 16777215                    ; 0xbaffffff
msr cpsr_c, 0                   ; 0xe328f000
bgt 16777215                    ; 0xcaffffff
msr cpsr_c, 521                 ; 0xe328f209
bgt 16777215                    ; 0xcaffffff
msr cpsr_c, 257                 ; 0xe328f101
ble 16777215                    ; 0xdaffffff
msr cpsr_c, 258                 ; 0xe328f102
ble 16777215                    ; 0xdaffffff
msr cpsr_c, 513                 ; 0xe328f201
ble 16777215                    ; 0xdaffffff
b 16777215                      ; 0xeaffffff